tests/posix/sem/prj.conf

```diff
@@ -0,0 +1,4 @@
+CONFIG_PTHREAD_IPC=y
+CONFIG_MAIN_THREAD_PRIORITY=2
+CONFIG_ZTEST=y
+CONFIG_SEM_VALUE_MAX=65535
```

Comment:
Max sem value can only be 32K, not 64K. please see my comment on last review.

Reply:
Also it will be better to change directory name of test to semaphore instead of sem.